api/src/auth/utils/is-login-redirect-allowed.ts

```diff
@@ -37,7 +36,5 @@ export function isLoginRedirectAllowed(redirect: unknown, provider: string): boo
 	}
 
 	// allow redirects to the defined PUBLIC_URL
-	const { protocol: publicProtocol, hostname: publicDomain } = new URL(publicUrl);
-
-	return `${redirectProtocol}//${redirectDomain}` === `${publicProtocol}//${publicDomain}`;
+	return redirectOrigin === new URL(publicUrl).origin;
```

Comment:
We should stick with checking the protocol + domain for origin as the `URL` origin will return the port which is not intentional nor something we enforce currently